assets/js/blocks/cart-checkout/subscriptions-integration.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import {
+	ExperimentalOrderShippingPackages,
+	ShippingRatesControlPackage,
+} from '@woocommerce/blocks-checkout';
+import { registerPlugin } from '@wordpress/plugins';
+import { useStoreCart, useSelectShippingRate } from '@woocommerce/base-hooks';
+import { useMemo } from '@wordpress/element';
+
+const RenderSubscriptionPackages = () => {
+	const {
+		extensions: { subscriptions },
+	} = useStoreCart();
+
+	// Flatten all packages from recurring carts.
+	const packages = useMemo( () => {
+		const newPackages = [];
+
+		Object.values( subscriptions ).forEach( ( recurringCart ) => {
+			const recurringCartPackages = recurringCart.shipping_rates || [];
+
+			recurringCartPackages.forEach( ( recurringCartPackage ) => {
+				newPackages.push( recurringCartPackage );
+			} );
+		} );
+
+		return newPackages;
+	}, [ subscriptions ] );
```

Comment:
This felt really hard for me to read, it can be simplified to this
```suggestion
const packages = useMemo(
		() =>
			Object.values( subscriptions )
				.map( ( recurringCart ) => recurringCart.shipping_rates )
				.filter( Boolean )
				.flat(),
		[ subscriptions ]
	);
```